src/shared/sleep-config.c

```diff
@@ -108,77 +111,187 @@ int parse_sleep_config(SleepConfig **ret_sleep_config) {
+
+        assert(ret);
+
+        current_capacity = hashmap_new(&string_hash_ops_free);
+        if (!current_capacity)
+                return log_oom();
+
+        r = battery_enumerator_new(&e);
         if (r < 0)
-               return log_debug_errno(r, "Failed to parse battery capacity: %m");
+                return log_error_errno(r, "Failed to initialize battery enumerator: %m");
 
-        if (battery_capacity < 0 || battery_capacity > 100)
-               return log_debug_errno(SYNTHETIC_ERRNO(ERANGE), "Invalid battery capacity");
+        FOREACH_DEVICE(e, dev) { /* For AC, the TYPE is mains */
+                const char *n, *battery_name;
 
-        return battery_capacity;
+                r = read_battery_capacity_percentage(dev);
+                if (r < 0) {
+                        log_debug_errno(r, "Failed to get battery capacity: %m");
+                        continue;
+                }
+                battery_capacity = r;
+
+                r = sd_device_get_property_value(dev, "POWER_SUPPLY_NAME", &battery_name);
+                if (r < 0) {
+                        log_device_debug_errno(dev, r, "Failed to read battery name: %m");
+                        continue;
+                }
+
+                n = strdup(battery_name);
+                if (!n)
+                        return log_oom();
+
+                r = hashmap_put(current_capacity, n, INT_TO_PTR(battery_capacity));
+                if (r < 0) {
+                        log_error_errno(r, "Failed to add to hashmap last battery capacity: %m");
+                        continue;
+                }
+        }
+        *ret = TAKE_PTR(current_capacity);
+
+        return 0;
 }
 
 /* Read file path and return hash of value in that file */
-static int get_battery_identifier(const char *filepath, struct siphash *ret) {
-        _cleanup_free_ char *value = NULL;
+static int get_battery_identifier(sd_device *dev, const char *property, struct siphash *state) {
+        /* _cleanup_free_ char *value = NULL; */
+        const char *x = NULL;
         int r;
 
-        assert(filepath);
-        assert(ret);
+        assert(dev);
+        assert(property);
+        assert(state);
 
-        r = read_one_line_file(filepath, &value);
+        r = sd_device_get_property_value(dev, property, &x);
         if (r == -ENOENT)
-               log_debug_errno(r, "%s is unavailable: %m", filepath);
+               log_device_debug_errno(dev, r, "battery device property is unavailable: %m");
         else if (r < 0)
-               return log_debug_errno(r, "Failed to read %s: %m", filepath);
-        else if (isempty(value))
-               log_debug_errno(SYNTHETIC_ERRNO(EINVAL), "%s is empty: %m", filepath);
+               return log_device_debug_errno(dev, r, "Failed to read battery device property: %m");
+        else if (isempty(x))
+               log_debug_errno(SYNTHETIC_ERRNO(EINVAL), "battery device property %s is null: %m", property);
```

Comment:
use log_device_debug_errno here too so that details about the specific device are printed

Reply:
fixed